src/translating/Translator.java

```diff
@@ -916,6 +926,19 @@ private void translateRule(ASTprogramRule rule,boolean writeWarningsToSTDERR) th
 		appendNewLineToTranslation();
 	}
 
+        private void processOptimizationStatement(ASTprogramRule rule) throws ParseException {
+            if(Settings.getSolver() == ASPSolver.Clingo) {
```

Comment:
Since you already checked for errors for DLV solver, it is better to have an assertion here.

You can say 

assert Settings.getSolver() == ASPSolver.Clingo;

There many reasons why it is better. Firstly, the if you added is never false. Hence if I use coverage analyzer on this code it will flag it. 

Secondly in future if I add a new solver, it will silently do a wrong thing. It is better for it to assert instead. Bad error is better than wrong numeric result!

Reply:
Changed the code to include this change.